nuxeo-connect-client/src/main/java/org/nuxeo/connect/connector/http/ConnectHttpConnector.java

```diff
@@ -110,19 +128,24 @@ protected ConnectServerResponse execServerCall(String url, Map<String, String> h
                 } catch (JSONException e) {
                     log.debug("Can't parse server error " + rc, e);
                     throw new ConnectServerError("Server returned a code " + rc);
+                } finally {
+                    httpResponse.close();
+                    httpClient.close();
                 }
             }
         } catch (ConnectServerError cse) {
-            method.releaseConnection();
             throw cse;
         } catch (Throwable t) {
-            String exName = t.getClass().getName();
-            if (exName.startsWith("java.net") || exName.startsWith("org.apache.commons.httpclient")) {
-                log.warn("Connect Server is not reachable");
-                method.releaseConnection();
-                throw new CanNotReachConnectServer(t.getMessage(), t);
+            try {
+                if (httpResponse != null) {
+                    httpResponse.close();
+                }
+                if (httpClient != null) {
+                    httpClient.close();
+                }
+            } catch (IOException e) {
+                log.warn("Unable to close Nuxeo Connect Server stream", e);
```

Comment:
You can use `IOUtils.closeQuietly` instead. I don't think there's a legitimate case where we'd be interested in the error log on `close`.